src/ApiRequestPanel.js

```diff
@@ -166,6 +168,14 @@ export class ApiRequestPanel extends EventsTargetMixin(HeadersParserMixin(LitEle
     `;
   }
 
+  _renderServerSelector() {
```

Comment:
So, what will happen when you select "custom" option in the selector and set an URL? It should update the `baseUri` on the components so they can handle this in the UI. But now when you do this the selector will disappear. This should not work like this. We need to know whether the base URI was an effect of setting it up in the selector or not. 
I guess that the right approach would be to add this information to the event dispatched by the selector and set some flag here. If the base URI is a result of setting it in the selector then the selector stays rendered. Otherwise it is removed. Also, the flag has to be cleared when you actually change the baseUri from the outside. It can't happen automatically but you can dispatch an event on the console to clear this state.

Reply:
This actually doesn't happen the way you're saying. When you select custom and set a URL, the `baseUri` property does not change, the `_selectedServer` property changes. This template function checks if the `baseUri` property is set (which is only being set by the parent), so the server selection change won't modify this value (i.e., I'm never reassigning `this.baseUri`)